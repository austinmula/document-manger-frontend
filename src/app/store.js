import { configureStore } from "@reduxjs/toolkit";
import counterReducer from "../features/counter/counterSlice";
import authReducer from "../features/auth/authSlice";
import filesReducer from "../features/files/filesSlice";
import roleReducer from "../features/roles/roleSlice";
import userReducer from "../features/users/userSlice";
import permissionReducer from "../features/permissions/permissionSlice";
import departmentReducer from "../features/departments/departmentSlice";

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    auth: authReducer,
    files: filesReducer,
    permissions: permissionReducer,
    roles: roleReducer,
    users: userReducer,
    departments: departmentReducer,
  },
});
