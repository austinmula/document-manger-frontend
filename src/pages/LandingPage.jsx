import React from "react";
import LandingContainer from "../components/landing/LandingContainer";

function LandingPage() {
  return <LandingContainer />;
}

export default LandingPage;
