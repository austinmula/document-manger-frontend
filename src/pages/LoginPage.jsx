import { UserCircleIcon } from "@heroicons/react/24/solid";
import LoginForm from "../components/forms/LoginForm";

export default function LoginPage() {
  return (
    <>
      <div className="w-full max-w-md space-y-8 px-4 py-6 bg-white rounded-md shadow-sm">
        <div>
          <UserCircleIcon className="mx-auto h-12 w-auto text-emerald-800" />

          <h2 className="mt-6 text-center text-3xl font-bold tracking-tight text-gray-700">
            Sign in to your account
          </h2>
          <p className="mt-2 text-center text-sm text-gray-600">
            Enter your email and password to access the dashboard.
          </p>
        </div>

        <LoginForm />
      </div>
    </>
  );
}
