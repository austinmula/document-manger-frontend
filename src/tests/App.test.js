import React from "react";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LandingPage from "../pages/LandingPage";
import LoginPage from "../pages/LoginPage";

test("take page snapshot and compare", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <LandingPage />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();

  const login = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <LoginPage />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(login).toMatchSnapshot();
});
