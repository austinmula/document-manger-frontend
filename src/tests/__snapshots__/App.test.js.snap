// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take page snapshot and compare 1`] = `
<div
  className="flex items-center justify-center h-screen"
>
  <div
    className="max-w-7xl w-10/12 "
  >
    <div
      className="grid grid-cols-1 md:grid-cols-2"
    >
      <div
        className="left w-full"
      >
        <img
          alt="File Manager"
          src="/File-manager.png"
        />
      </div>
      <div
        className="px-5 flex gap-5 flex-col"
      >
        <h1
          className="text-3xl md:text-5xl my-4 tracking-wide text-emerald-800 font-bold"
        >
          <span
            className="block my-5 whitespace-nowrap"
          >
            Flysystem
          </span>
          Document Manager
        </h1>
        <p
          className="font-thin text-sm md:text-lg text-gray-900"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Reprehenderit inventore rerum ut sunt adipisci labore. Facere magni laudantium delectus quasi illum repellat neque laboriosam,
        </p>
        <div>
          <button
            className="group relative flex justify-center rounded-md border border-transparent bg-emerald-800 py-4 p-4 text-sm font-medium text-white hover:bg-emerald-700 focus:outline-none focus:ring-2 focus:ring-emerald-500 focus:ring-offset-2"
            onClick={[Function]}
            type="submit"
          >
            Proceed To Login
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`take page snapshot and compare 2`] = `
<div
  className="w-full max-w-md space-y-8 px-4 py-6 bg-white rounded-md shadow-sm"
>
  <div>
    <svg
      aria-hidden="true"
      className="mx-auto h-12 w-auto text-emerald-800"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M18.685 19.097A9.723 9.723 0 0021.75 12c0-5.385-4.365-9.75-9.75-9.75S2.25 6.615 2.25 12a9.723 9.723 0 003.065 7.097A9.716 9.716 0 0012 21.75a9.716 9.716 0 006.685-2.653zm-12.54-1.285A7.486 7.486 0 0112 15a7.486 7.486 0 015.855 2.812A8.224 8.224 0 0112 20.25a8.224 8.224 0 01-5.855-2.438zM15.75 9a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0z"
        fillRule="evenodd"
      />
    </svg>
    <h2
      className="mt-6 text-center text-3xl font-bold tracking-tight text-gray-700"
    >
      Sign in to your account
    </h2>
    <p
      className="mt-2 text-center text-sm text-gray-600"
    >
      Enter your email and password to access the dashboard.
    </p>
  </div>
  <form
    className="mt-8 space-y-6"
    onSubmit={[Function]}
  >
    <div
      className="space-y-4 rounded-md shadow-sm"
    >
      <div>
        <label
          className="block font-bold text-sm mb-2 text-gray-600"
          htmlFor="email"
        >
          Email
        </label>
        <div
          className="form-field"
        >
          <input
            className="block w-full appearance-none rounded-md border  px-3 py-2  focus:outline-none border-gray-300 focus:border-indigo-400"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your Email"
            type="email"
          />
          <small
            className="text-red-300"
          />
        </div>
      </div>
      <div>
        <label
          className="block font-bold text-sm mb-2 text-gray-600"
          htmlFor="password"
        >
          Password
        </label>
        <div
          className="form-field"
        >
          <input
            className="block w-full appearance-none rounded-md border  px-3 py-2  focus:outline-none border-gray-300 focus:border-indigo-400"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
          />
          <small
            className="text-red-300"
          />
        </div>
      </div>
    </div>
    <div>
      <button
        className="group relative flex w-full justify-center rounded-md border border-transparent bg-emerald-800 py-2 px-4 text-sm font-medium text-white hover:bg-emerald-700 focus:outline-none focus:ring-2 focus:ring-emerald-500 focus:ring-offset-2"
        disabled={false}
        type="submit"
      >
        <span
          className="absolute inset-y-0 left-0 flex items-center pl-3"
        >
          <svg
            aria-hidden="true"
            className="h-5 w-5 text-emerald-700 group-hover:text-emerald-600"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M12 1.5a5.25 5.25 0 00-5.25 5.25v3a3 3 0 00-3 3v6.75a3 3 0 003 3h10.5a3 3 0 003-3v-6.75a3 3 0 00-3-3v-3c0-2.9-2.35-5.25-5.25-5.25zm3.75 8.25v-3a3.75 3.75 0 10-7.5 0v3h7.5z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        Sign in
      </button>
    </div>
  </form>
</div>
`;
